elide-core/src/main/java/com/yahoo/elide/graphql/GraphQLRequestScope.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2017, Oath Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+package com.yahoo.elide.graphql;
+
+import com.yahoo.elide.ElideSettings;
+import com.yahoo.elide.core.DataStoreTransaction;
+import com.yahoo.elide.core.RequestScope;
+import com.yahoo.elide.security.User;
+import lombok.Getter;
+
+import javax.ws.rs.core.MultivaluedHashMap;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Persistent state across GraphQL requests. This object is stored in the environment context.
+ */
+public class GraphQLRequestScope extends RequestScope {
+    @Getter private final Map<String, Long> totalRecordCounts = new HashMap<>();
+
+    public GraphQLRequestScope(DataStoreTransaction transaction,
+                               User user,
+                               MultivaluedHashMap<String, String> queryParams,
```

Comment:
Is this needed?  THe endpoint passes in null for this.

Reply:
I left it open for the future-- presumably we'll want to provide access to headers in the future.